gobblin-data-management/src/main/java/org/apache/gobblin/data/management/policy/NewestKSelectionPolicy.java

```diff
@@ -61,6 +63,12 @@ public class NewestKSelectionPolicy<T extends DatasetVersion> implements Version
    */
   public static final String NEWEST_K_VERSIONS_NOTSELECTED_KEY = "selection.newestK.versionsNotSelected";
 
+  /**
+   * This denotes to use iterator for fetching all the versions of dataset.
+   * If true, this can't be used along with NewestKSelectionPolicy
+   */
+  public static final String VERSION_FINDER_ITERATOR = "version.finder.iterator";
```

Comment:
lets name it interrogative type , how about SHOULD_ITERATE_VERSIONS ?

Reply:
Done